pdns/ws-auth.cc

```diff
@@ -1390,7 +1390,20 @@ static void apiServerZones(HttpRequest* req, HttpResponse* resp) {
     throw HttpMethodNotAllowedException();
 
   vector<DomainInfo> domains;
-  B.getAllDomains(&domains, true); // incl. disabled
+
+  auto zone_it = req->getvars.find("zone");
+  if (zone_it != req->getvars.end()) {
```

Comment:
I think you can check `req->getvars.count("zone") != 0` and then use `req->getvars["zone"]`.

Reply:
Yeah, but that sounded like doing the work twice?

Reply:
That's just that something like the following looks better IMHO :
```
if (req->getvars.count("zone") != 0) {
  auto zone = req->getvars["zone"];
}
```

Reply:
Pushed the style change.